src/huggingface_hub/hf_api.py

```diff
@@ -318,13 +318,17 @@ def login(self, username: str, password: str) -> str:
         write_to_credential_store(username, password)
         return d["token"]
 
-    def whoami(self, token: str) -> Dict:
+    def whoami(self, token: Optional[str] = None) -> Dict:
         """
         Call HF API to know "whoami".
 
         Args:
-            token (``str``): Hugging Face token.
+            token (``str``, `optional`):
+                Hugging Face token. Will default to the locally saved token if not provided.
         """
+        if token is None:
+            token = HfFolder.get_token()
+
```

Comment:
WDYT @osanseviero? I think this is a great improvement in terms of user experience and should also be used to reopen conversations about:
- having the `HfApi.login` saving the token
- having the token default to the locally saved token when not passed through the `HfApi`

Reply:
Agree it's a great improvement! Thanks @sgugger! (related issue: #154)